third_party/terraform/tests/data_google_game_services_game_server_deployment_rollout_test.go.erb

```diff
@@ -0,0 +1,108 @@
+<% autogen_exception -%>
+package google
+
+<% unless version == 'ga' -%>
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"testing"
+)
+
+func TestAAccDataSourceGameServicesGameServerDeploymentRollout_basic(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"random_suffix": randString(t, 10),
+	}
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProvidersOiCS,
+		CheckDestroy: testAccCheckGameServicesGameServerDeploymentRolloutDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceGameServicesGameServerDeploymentRollout_basic(context),
+				Check: resource.ComposeTestCheckFunc(
+					testAccDataSourceCheck("data.google_game_services_game_server_deployment_rollout.qa", "google_game_services_game_server_deployment_rollout.foo"),
```

Comment:
Instead of this custom check, could you use `checkDataSourceStateMatchesResourceState`?